File: coding-test/프로그래머스/level1/완주하지못한선수/완주하지못한선수.js
function solution(participant, completion) {
  const obj = {};
  for (let player of participant) {
    if (obj[player]) {
      obj[player] += 1;
    } else {
      obj[player] = 1;
    }
  }
  for (let finisher of completion) {
    if (obj[finisher]) {
      obj[finisher] -= 1;
    }
  }
  for (let player of participant) {
    if (obj[player] >= 1) {
      return player;
    }
  }
}

File: coding-test/프로그래머스/level1/완주하지못한선수/완주하지못한선수.py
def solution(participants, completions):
    names = {}
        
    for participant in participants:
        if participant in names:
            names[participant] += 1
        else:
            names[participant] = 1
    
    for completion in completions:   
        names[completion] -= 1
        
             
    for player in participants:
        if names[player] > 0:
            return player



# 다른 사람 풀이

import collections


def solution(participant, completion):
    answer = collections.Counter(participant) - collections.Counter(completion)
    return list(answer.keys())[0]